### PR TITLE
ADR list - sorting and filtering

### PR DESCRIPTION
**Pull request by PGaigole**
- The ADR's listing to be sortable by the users (sorting by title & date)
- Only accepted ADRs are shown in the default view.
- The ADR listing to be filterable by the users (status)

### CODE REVIEW

File: assets/adr.scss [hunk 1]
```diff
@@ -67,14 +67,47 @@
...
-      display: flex;
-      flex-wrap: wrap;
-      column-gap: 10px;
-      padding: 0;
-      .filter-button {
+  .adr-controls {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    .filter-section,
```
Comment (swarad07) [inline assets/adr.scss:74]: Add line break here

File: assets/adr.scss [hunk 2]
```diff
@@ -67,14 +67,47 @@
...
+        list-style: none;
+        display: flex;
+        flex-wrap: wrap;
+        column-gap: 10px;
+        padding: 0;
+        .filter-item {
+          margin: 0;
+        }
+      }
+      .sort-list,
```
Comment (swarad07) [inline assets/adr.scss:86]: line break here

File: assets/adr.scss [hunk 3]
```diff
@@ -67,14 +67,47 @@
...
+        z-index: 1;
+        .sort-item,
+        .status-filter-item {
+          cursor: pointer;
+          &.active {
+            font-weight: bold;
+          }
+        }
+      }
+      .status-filter-list {
```
Comment (swarad07) [inline assets/adr.scss:106]: Possible to add stylelint here so that formatting for SCSS is taken care off?

Comment (PGaigole) [reply assets/adr.scss:106]: @swarad07 Thank you for the suggestion. I'll be using stylelint henceforth.

File: assets/js/sort-adr.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+function onToggle(elementId) {
+  document.getElementById(elementId).classList.toggle('hidden');
+}
+
+function sortBy(sortType) {
+  // Select all elements with class 'adr-links' and convert NodeList to array
+  let adrEntries = Array.from(
+    document.getElementById('adr-links').querySelectorAll('a'),
```
Comment (zeshanziya) [inline assets/js/sort-adr.js:8]: We can avoid redundant calls to `document.getElementById('adr-links')`. We can store it in a variable and use it in other places.

`const adrContainer = document.getElementById('adr-links');`

**Review by zeshanziya — COMMENTED** [review]
@PGaigole If we click on "Sort By" button followed by the "Filter By" button and then "Sort By" again, it doesn't work, you can add logic to hide the other dropdown when one is clicked. This ensures that only one dropdown is visible at a time.

File: assets/js/sort-adr.js (hunk 4 above)
Comment (PGaigole) [reply assets/js/sort-adr.js:8]: Thank you @zeshanziya for the review. I'll make necessary changes in upcoming commits.

**Comment by swarad07** [comment]
All well on the CSS front now. Will let Zeshan review the functional part and merge

File: assets/js/sort-adr.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        return dateA - dateB;
+      });
+      break;
+    default:
+      // No sorting action needed
+      break;
+  }
+
+  // Clear existing cards
+  adrLinks.innerHTML = '';
```
Comment (zeshanziya) [inline assets/js/sort-adr.js:61]: Sorting and filtering may take some time as the number of ADRs increases. I suggest setting the innerHTML to `Loading... Please wait` or a similar message at the top before sorting or filtering starts.

File: assets/js/sort-adr.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  // Append sorted cards back to the list
+  adrEntries.forEach((card) => {
```
Comment (zeshanziya) [inline assets/js/sort-adr.js:64]: Please add a message when `adrEntries` list is empty

`<p>No ADR pages found.</p>`

**Review by zeshanziya — COMMENTED** [review]
We should hide the dropdown upon selection or when clicking somewhere outside. Currently, it remains open until those buttons are clicked again.

**Review by zeshanziya — CHANGES_REQUESTED** [review]
@PGaigole Thanks for all the work. Please see some observations:

- Clicking the category buttons opens the filter dropdown.
- The filter should work within the current tab. So if I am under the `Drupal` category, it should filter from that category only. Currently, it switches to `All` when we filter the results.

If adding a loader seems complicated, we can leave it for now and worry about it later when it actually matters.

**Comment by PGaigole** [comment]
> @PGaigole Thanks for all the work. Please see some observations:
> 
> * Clicking the category buttons opens the filter dropdown.
> * The filter should work within the current tab. So if I am under the `Drupal` category, it should filter from that category only. Currently, it switches to `All` when we filter the results.
> 
> If adding a loader seems complicated, we can leave it for now and worry about it later when it actually matters.

I will take a look at these observations and work on them. Currently, I added a loader, I think if choose a slower connection, you might be able to see the same.

**Review by zeshanziya — APPROVED** [review]
Thanks @PGaigole , Its working fine now.

**Comment by hussainweb** [comment]
This is good work but few things can be made better. Let's do those in a follow-up.

**Pull request merged**